tools/chapel-py/src/core-types-gen.h

```diff
@@ -0,0 +1,92 @@
+/*
+ * Copyright 2023-2024 Hewlett Packard Enterprise Development LP
+ * Other additional copyright holders may be indicated within.
+ *
+ * The entirety of this work is licensed under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ *
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#ifndef CHAPEL_PY_CORE_TYPES_GEN_H
+#define CHAPEL_PY_CORE_TYPES_GEN_H
+
+#include "core-types.h"
+#include "error-tracker.h"
+#include "chpl/uast/all-uast.h"
+#include "python-types.h"
+
+/**
+  Declare a Python PyTypeObject that corresponds to a generated type
+  (AST node, Chapel type, etc.) of a given name.
+ */
+#define DECLARE_PY_OBJECT_FOR(ROOT, NAME)\
```

Comment:
nit, this macro should probably be cleaned up after its used as `#undef DECLARE_PY_OBJECT_FOR`

Reply:
Not sure what you mean. It's undef'ed on line 47, right after it's used in:

```C++
/* Generate a Python object for reach AST node type. */
#define GENERATED_TYPE(ROOT, NAME, TAG, FLAGS) DECLARE_PY_OBJECT_FOR(ROOT, NAME)
#include "generated-types-list.h"
#undef DECLARE_PY_OBJECT_FOR
```